### ISSUE TITLE
Uncaught Error: document is not defined

### ISSUE THREAD

**Issue by nolanlawson** [naptha/tesseract.js#352]
Thank you for tesseract.js! :slightly_smiling_face:

**Describe the bug**

When using the [tesseract.js-offline](https://github.com/jeromewu/tesseract.js-offline/) steps with the latest versions (`"tesseract.js": "^2.0.0-beta.1", "tesseract.js-core": "^2.0.0-beta.13"`), an error is thrown:

    Uncaught Error: ReferenceError: document is not defined

The line that appears to be throwing is [this one](https://github.com/lydell/resolve-url/blob/c99e0ad30108ccdf73e10eddbe7c8b2080c57740/resolve-url.js#L21) inside of the `resolve-url` library. It seems tesseract.js is trying to resolve a URL, but it fails because it's inside of a worker and so `document` is undefined.

**To Reproduce**

I created a minimal repro in [this gist](https://gist.github.com/nolanlawson/111e12042cdc6e30581d88b764b56275). You can repro with the following steps:

```sh
git clone git@gist.github.com:111e12042cdc6e30581d88b764b56275.git repro
cd repro
npm i
npx webpack
npx hs -p 8080
# then open up localhost:8080 in your browser
```

**Expected behavior**

It should not throw an error.

**Screenshots**

![Screenshot from 2019-10-20 09-19-36](https://user-images.githubusercontent.com/283842/67162646-c1f0e500-f31a-11e9-8b4a-5202ee5f5404.png)

**Desktop:**
 - OS: Ubuntu
 - Browser Chrome
 - Version 77

**Context:**

I'm trying to upgrade from an older alpha version (`"tesseract.js": "^2.0.0-alpha.13", "tesseract.js-core": "^2.0.0-beta.10"`). The older version is working great. Looking forward to the upgrade!

**Comment by jeromewu** [comment]
Hi, it is kind of weird but in my environment is works with your gist and I am using Ubuntu with Chrome 77 as well.

![Screenshot from 2019-10-21 23-48-15](https://user-images.githubusercontent.com/5723124/67221003-582d2400-f45d-11e9-89f0-be3af2e66e5e.png)

**Comment by nolanlawson** [comment]
I just realized the difference &ndash; apparently Tesseract has different behavior depending on whether you are on `localhost`/`127.0.0.1` versus a custom hostname:

![out](https://user-images.githubusercontent.com/283842/67295340-d4def300-f49b-11e9-9eab-6654d59c5ddd.png)

You can see in the screenshot above that `localhost`/`127.0.0.1` work fine, but not `lolcathost`. (I've added `lolcathost` to my `/etc/hosts` to map to `127.0.0.1`, and I use it occasionally for local testing because it automatically disallow Service Worker… guess I forgot I was using it instead of `localhost`. :slightly_smiling_face:)

Maybe this is still a worthwhile bug to pursue, because it's the kind of bug that will pop up in production but not locally? I'm not sure if there's a difference between `https` vs `http` though.

**Comment by jeromewu** [comment]
I do a double check and confirm there is a bug, I have fixed it and release 2.0.0-beta.2, please kindly check.

**Comment by abhishek-appsdeployer** [comment]
[ReferenceError: Property 'document' doesn't exist] , I got this error in using tesseract.js in react native.
